### DISCUSSION TITLE
Minio IAM AWS compatibility

### DISCUSSION THREAD

**Discussion by avinashs433** [minio/minio#16082] [Q&A] [answered]
Hi,

I am able to use the AWS s3 SDK to communicate with Minio. Does minio support the AWS IAM client as well to manage users, groups and policies against Minio ?

I have tried using the 'AmazonIdentityManagementServiceClient' and get the following error:
"Can not get credentials, api error: Invalid STS API version 2010-05-08, expecting 2011-06-15"

```
var awsCredentials = new BasicAWSCredentials(_options.Value.S3Service.Username, _options.Value.S3Service.Password);
var identityManagementServiceConfig = new AmazonIdentityManagementServiceConfig()
{
     ServiceURL = endpoint.OriginalString,
    UseHttp = _options.Value.S3Service.SSL,
    ProxyHost = proxy.Host,
    ProxyPort = proxy.Port
};

_awsIMClient = new AmazonIdentityManagementServiceClient(awsCredentials, identityManagementServiceConfig);
var users = await _awsIMClient.ListUsersAsync();
```

Regards,
Avinash

**Comment by harshavardhana** [comment, marked as answer]
We only implement S3 APIs, IAM APIs are not part of MinIO server.

**Reply by avinashs433** [reply]
Cool - thanks

**Reply by bendem** [reply]
Could this be considered? It would make provisioning a minio cluster a lot simpler (reusing the same playbooks and code across s3 providers).

**Reply by harshavardhana** [reply]
@bendem sure if you have a large enough requirement and there is commercial interest reach out to us at hello@minio.io 

Without that, there is no real incentive for us to carry this baggage.
